### PR TITLE
Add CRD reference docs

### PR DESCRIPTION
**Pull request by randomvariable**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

* Adds CRD documentation for Cluster API AWS and clusterawsadm
* Optimises book generation a little

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits
- [X] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/book/Makefile [hunk 1]
```diff
@@ -20,30 +20,45 @@ MDBOOK := $(TOOLS_BIN_DIR)/mdbook
...
 
+src/crd:
+	mkdir -p src/crd
+
+src/crd/cluster-api-aws.md: $(GENCRDAPIREFERENCEDOCS) src/crd $(API_SRCS) $(GENCRDAPIREFERENCEDOCS_SRCS) $(BOOK_SRCS)
+	$(GENCRDAPIREFERENCEDOCS) -template-dir gen-crd-api-reference-docs/template -config "gen-crd-api-reference-docs/config.json" \
+		-api-dir "../../api/v1alpha3" \
+		-out-file $@
+
+src/crd/clusterawsadm.md: $(GENCRDAPIREFERENCEDOCS) src/crd $(API_SRCS) $(GENCRDAPIREFERENCEDOCS_SRCS) $(BOOK_SRCS)
```
Comment (sedefsavas) [inline docs/book/Makefile:41]: Do we need API_SRCS and BOOK_SRCS? 
```suggestion
src/crd/clusterawsadm.md: $(GENCRDAPIREFERENCEDOCS) src/crd $(GENCRDAPIREFERENCEDOCS_SRCS) 
```
```suggestion
src/crd/clusterawsadm.md: $(GENCRDAPIREFERENCEDOCS) src/crd $(API_SRCS) $(GENCRDAPIREFERENCEDOCS_SRCS) $(BOOK_SRCS)
```

Comment (randomvariable) [reply docs/book/Makefile:41]: We do because we want the CRD docs to update when the API changes.

**Pull request merged**